src/lib/RangeHelper.js

```diff
@@ -385,6 +385,82 @@ define(function (require) {
 			return '';
 		};
 
+
+		/**
+		 * Replace elements with provided selector inside user selection with
+		 * their children
+		 *
+		 * This method provides an easier way to have the same final aspect
+		 * for HTML as powerful as the browser's native execCommand() method
+		 * when removing HTML tag formatting.
+		 *
+		 * This method is mostly meant to be used with <b>inline</b> HTML
+		 * Elements. Having it working with block level elements is not
+		 * guaranteed.
+		 *
+		 *
+		 * @param {String} removeElementSelector Selector match what to delete
+		 * @return undefined
+		 * @function
+		 * @name splitAtSelection
+		 * @memberOf RangeHelper.prototype
+		 * @author brunoais
+		 */
+		base.splitAtSelection = function (removeElementSelector) {
```

Comment:
The one I could reproduce best was using firefox 37.0.2 with a collapsed selection, after selection is restored after the split, the bold inline tag was being removed until the end of the parent block tag instead of just at collapsed selection.
Some others were hard to reproduce and I don't even know if they can be simplified.